compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -2761,6 +2761,11 @@ private void checkInLangLib(BLangInvocation iExpr, BType varRefType) {
         }
 
         checkIllegalStorageSizeChangeMethodCall(iExpr, varRefType);
+
+        // check argument types in arr:sort function
+        if (iExpr.langLibInvocation && iExpr.name.value.equals("sort")) {
```

Comment:
At this point, don't need the `iExpr.langLibInvocation` check here since at this point, we know for sure (Line# 2753) that this is a lang lib call.